app/code/Magento/ConfigurableProduct/Plugin/Model/Attribute/Backend/AttributeValidation.php

```diff
@@ -29,27 +33,41 @@ public function __construct(
     }
 
     /**
-     * @param \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend $subject
+     * Verify is attribute used for configurable product creation and should not be validated.
+     *
+     * @param AbstractBackend $subject
      * @param \Closure $proceed
-     * @param \Magento\Framework\DataObject $entity
+     * @param DataObject $entity
      * @return bool
      */
     public function aroundValidate(
-        \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend $subject,
+        AbstractBackend $subject,
         \Closure $proceed,
-        \Magento\Framework\DataObject $entity
+        DataObject $entity
     ) {
         $attribute = $subject->getAttribute();
-        if ($entity instanceof ProductInterface
-            && $entity->getTypeId() == Configurable::TYPE_CODE
-            && in_array(
-                $attribute->getAttributeId(),
-                $this->configurableProductType->getUsedProductAttributeIds($entity),
-                true
-            )
-        ) {
+        if ($this->isAttributeShouldBeValidated($entity, $attribute)) {
             return true;
         }
         return $proceed($entity);
     }
+
+    /**
+     * Verify if attribute is a part of configurable product and should not be validated.
+     *
+     * @param DataObject $entity
+     * @param AbstractAttribute $attribute
+     * @return bool
+     */
+    private function isAttributeShouldBeValidated(DataObject $entity, AbstractAttribute $attribute): bool
+    {
+        if (!($entity instanceof ProductInterface && $entity->getTypeId() == Configurable::TYPE_CODE)) {
```

Comment:
Its better use strict comparison is possible

Reply:
Fixed